File: src/main/java/my/study/repository/design/pattern/flyweight/TicketImp.java
package my.study.repository.design.pattern.flyweight;

/**
 * @author djh on  2018/9/19 22:16
 * @E-Mail 1544579459@qq.com
 */
public class TicketImp implements Ticket {

    private String fromPlace;
    private String toPlace;
    private int price;

    TicketImp(String fromPlace, String toPlace, int price) {
        this.fromPlace = fromPlace;
        this.toPlace = toPlace;
        this.price = price;
    }

    @Override
    public void showTicketInfo(String bunk) {
        System.out.println("起点站:" + fromPlace + " 终点站:" + toPlace + " 价格:" + price + " 铺位:" + bunk);
    }
}

File: src/main/java/my/study/repository/design/pattern/flyweight/kotlin/TicketImp.kt
package my.study.repository.design.pattern.flyweight.kotlin

/**
 * @author djh on  2018/10/5 19:04
 * @E-Mail 1544579459@qq.com
 */
class TicketImp(
        private val price: Int,
        private val toPlace: String,
        private val fromPlace: String) : Ticket {
    
    override fun showTicketInfo(bunk: String) {
        println("起点站: $fromPlace -> 终点站: $toPlace, 铺位: $bunk, 价格: $price")
    }
}